engine/objconv/basemaker/base_maker.cpp

```diff
@@ -2166,8 +2166,8 @@ int main(int argc, char **argv) {
     Base::CurrentBase = NULL;
     glutInit(&argc, argv);
     glutInitWindowSize(800, 600);
-    g_game.x_resolution = 800;
-    g_game.y_resolution = 600;
+    g_game.x_resolution = 1920;
+    g_game.y_resolution = 1080;
```

Comment:
These should probably be in variables/constants and match L2168. (1920, 1080) is outside the window defined by L2168.

Reply:
The plan (fairly advanced actually) is to move all settings in the engine to an automatically generated and populated singleton. That would deprecate a huge chunk of code, from settings, configuration, all those XMLParser lines, etc.

The JSON parser would convert directly to the filetype and you'd work against a variable. https://github.com/royfalk/JsonStruct has the script to generate this.

Reply:
@royfalk that's fine - just calling out the potential bug due to the the window size not matching here.

Reply:
That's interesting. I guess I need to figure out what objconv does. It's not part of the main game,as far as I can tell. I'm also not convinced we are using glutInit in the actual game. I think both SDL and glut were supported and I picked SDL.

This lack of focus by the original devs... well.

Reply:
At least when working with Qt, the coordinate system worked in two parts:
1. Canvas that everything was drawn against
2. View Port for what on the canvas to show the user
3. Device Canvas that laid out the hardware

It gave some interesting effects. So technically, those coordinates exist but they may be outside the view port; but that's probably not what we want.